integration-tests/src/tests/client/features/stateless_validation.rs

```diff
@@ -225,25 +230,46 @@ fn run_chunk_validation_test(seed: u64, prob_missing_chunk: f64) {
     if prob_missing_chunk >= 0.8 {
         assert!(found_differing_post_state_root_due_to_state_transitions);
     }
+
+    // print some extra info to easy debugging
+    let genesis_height = genesis.config.genesis_height;
+    for i in 0..blocks_to_produce {
+        let client = &env.clients[0];
+        let height = genesis_height + i as u64;
+        let block = client.chain.get_block_by_height(height);
+        let Ok(block) = block else {
+            tracing::info!(target: "test", "Block {}: missing", height);
+            continue;
+        };
+        let prev_hash = block.header().prev_hash();
+        let epoch_id = client.epoch_manager.get_epoch_id_from_prev_block(prev_hash).unwrap();
+        let protocol_version = client.epoch_manager.get_epoch_protocol_version(&epoch_id).unwrap();
+
+        tracing::info!(target: "test", "Block {}-{}: {} -> {}", height, block.hash(), protocol_version, block.header().latest_protocol_version());
+    }
 }
```

Comment:
actually one more comment, do we check automatically (other than looking at these logs) that the protocol upgrade actually happened? or is there something else that implies it?

Reply:
Ah, thanks, good point I was going to do that and I forgot.